src/components/calendar/CalendarConnected.tsx

```diff
@@ -38,25 +38,28 @@ const mapStateToProps = (state: IReactVaporState, ownProps: ICalendarOwnProps):
 const mapDispatchToProps = (dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
                             ownProps: ICalendarOwnProps): ICalendarDispatchProps => ({
     onClick: (pickerId: string, isUpperLimit: boolean, value: Date) => {
-      dispatch(selectDate(pickerId, ''));
-      dispatch(changeOptionPicker(pickerId, '', ''));
-      if (!value) {
-        dispatch(resetDatePickers(pickerId));
-      } else {
+      dispatch(resetOptionPickers(pickerId));
+      if (value) {
         if (isUpperLimit) {
           dispatch(changeDatePickerUpperLimit(pickerId, moment(value).endOf('day').toDate()));
         } else {
           dispatch(changeDatePickerLowerLimit(pickerId, value));
-          dispatch(selectDate(pickerId, DateLimits.upper));
 
           // mirror upper limit to lower limit if not linked with a date range
           // this will cause the selected lower limit date to display in the calendar right after selection of the lower limit date
-          if (!ownProps.isLinkedToDateRange && ownProps.isLinkedToDateRange !== undefined) {
+          if (!_.isUndefined(ownProps.isLinkedToDateRange) && !ownProps.isLinkedToDateRange) {
             dispatch(changeDatePickerUpperLimit(pickerId, value));
+            dispatch(selectDate(pickerId, DateLimits.lower));
           }
         }
+      } else {
+        dispatch(resetDatePickers(pickerId));
       }
```

Comment:
not too much a fan of nested ifs, i know it's debatable but, here's my suggestion:

```
if (value && isUpperLimit) {
    dispatch(changeDatePickerUpperLimit(pickerId, moment(value).endOf('day').toDate()));
} else if (value) {
    dispatch(changeDatePickerLowerLimit(pickerId, value));

    // mirror upper limit to lower limit if not linked with a date range
    // this will cause the selected lower limit date to display in the calendar right after selection of the lower limit date
    if (!_.isUndefined(ownProps.isLinkedToDateRange) && !ownProps.isLinkedToDateRange) {
        dispatch(changeDatePickerUpperLimit(pickerId, value));
        dispatch(selectDate(pickerId, DateLimits.lower));
    }
} else {
    dispatch(resetDatePickers(pickerId));
}
```

Reply:
Personnaly, I don't have a preference so I'll apply your solution, thanks.

Reply:
Fans of early return?